packages/core/theme-module/theme/pages/Category.vue

```diff
@@ -396,88 +257,74 @@ const sortByOptions = [
-    { label: 'Black', value: 'black', color: '#000000', selected: false },
-    { label: 'Yellow', value: 'yellow', color: '#DCA742', selected: false },
-    { label: 'Blue', value: 'blue', color: '#004F97', selected: false },
-    { label: 'Navy', value: 'navy', color: '#656466', selected: false }
-  ],
-  size: [
-    { label: 'Size 2 (XXS)', value: 'xxs', count: '10', selected: false },
-    { label: 'Size 4-6 (XS)', value: 'xs', count: '23', selected: false },
-    { label: 'Size 8-10 (S)', value: 's', count: '54', selected: false },
-    { label: 'Size 12-14 (M)', value: 'm', count: '109', selected: false },
-    { label: 'Size 16-18 (L)', value: 'l', count: '23', selected: false },
-    { label: 'Size 20-22(XL)', value: 'xl', count: '12', selected: false },
-    { label: 'Size 24-26 (XXL)', value: 'xxl', count: '2', selected: false }
-  ],
-  price: [
-    { label: 'Under $200', value: 'under-200', count: '23', selected: false },
-    { label: 'Under $300', value: 'under-300', count: '54', selected: false }
-  ],
-  material: [
-    { label: 'Cotton', value: 'coton', count: '33', selected: false },
-    { label: 'Silk', value: 'silk', count: '73', selected: false }
-  ]
-};
-
 // TODO: to be implemented in https://github.com/DivanteLtd/next/issues/211
 const breadcrumbs = [
   { text: 'Home', route: { link: '#' } },
   { text: 'Women', route: { link: '#' } }
 ];
 
-function updateFilter() {}
-
-function clearAllFilters() {
-  const filtersNames = Object.keys(filters);
-  filtersNames.forEach((name) => {
-    filters[name].forEach((value) => {
-      value.selected = false;
-    });
-  });
-}
-
 export default {
   transition: 'fade',
   setup(props, context) {
     const { query } = context.root.$route;
 
     const { categories, search, loading } = useCategory('categories');
-    const { products: categoryProducts, totalProducts, search: productsSearch, loading: productsLoading } = useProduct('categoryProducts');
+    const {
+      products: categoryProducts,
+      totalProducts,
+      search: productsSearch,
+      loading: productsLoading,
+      availableFilters
+    } = useProduct('categoryProducts');
+
     const currentPage = ref(parseInt(query.page, 10) || 1);
     const itemsPerPage = ref(parseInt(query.items, 10) || perPageOptions[0]);
+    const filters = ref(null);
+
+    const filtersFromQuery = Object.entries(query).filter(([name]) => !['page', 'items'].includes(name));
+
+    const productsSearchParams = computed(() => ({
+      catId: (categories.value[0] || {}).id,
+      page: currentPage.value,
+      perPage: itemsPerPage.value,
+      filters: filters.value
+    }));
 
     onSSR(async () => {
       await search(getCategorySearchParameters(context));
-      await productsSearch({
-        catId: (categories.value[0] || {}).id,
-        page: currentPage.value,
-        perPage: itemsPerPage.value
+      await productsSearch(productsSearchParams.value);
+      filters.value = availableFilters.value;
```

Comment:
create  a dedicated function called `getFitersFromSearchParameters`  to build filters based on a search query justly like `getCategorySearchParameters` for searching params

Reply:
it can be different for each platform, so always you will be able to override that function